core/templates/dev/head/domain/feedback_thread/FeedbackThreadObjectFactory.js

```diff
@@ -0,0 +1,84 @@
+// Copyright 2015 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for creating new frontend instances of feedback thread
+   domain objects.
+ */
+
+oppia.factory('FeedbackThreadObjectFactory', [function() {
+  var FeedbackThread = function(status, originalAuthorId, lastUpdated,
+    lastMessageText, totalNoOfMessages, lastMessageRead,
+    secondLastMessageRead, authorLastMessage, authorSecondLastMessage,
+    explorationTitle, explorationId, threadId) {
+    this.status = status;
+    this.originalAuthorId = originalAuthorId;
+    this.lastUpdated = lastUpdated;
+    this.lastMessageText = lastMessageText;
+    this.totalNoOfMessages = totalNoOfMessages;
+    this.lastMessageRead = lastMessageRead;
+    this.secondLastMessageRead = secondLastMessageRead;
+    this.authorLastMessage = authorLastMessage;
+    this.authorSecondLastMessage = authorSecondLastMessage;
+    this.explorationTitle = explorationTitle;
+    this.explorationId = explorationId;
+    this.threadId = threadId;
+  };
+
+  FeedbackThread.prototype.updateSummaryOnNewMessage = function(
```

Comment:
This function was intended to update the summaries of the thread on the addition of a new message. Now the function name has been changed to addNewMessage. So now it reads threadSummary.addNewMessage which I think gives the intended meaning.